Sources/NIO/ChannelPipeline.swift

```diff
@@ -400,11 +400,20 @@ public final class ChannelPipeline: ChannelInvoker {
             promise?.fail(ChannelError.unremovableHandler)
             return
         }
-        if self.eventLoop.inEventLoop {
+        func removeHandler0() {
+            guard !context.userTriggeredRemovalStarted else {
+                promise?.fail(NIOAttemptedToRemoveHandlerMultipleTimesError())
+                return
+            }
+            context.userTriggeredRemovalStarted = true
             handler.removeHandler(context: context, removalToken: .init(promise: promise))
+        }
```

Comment:
Given that his function touches `context` a few times, would it be better to have this be a function on `ChannelHandlerContext` instead? That would also allow `userTriggeredRemovalStarted` to be a `private var` instead of an `internal var`.

Reply:
@Lukasa sounds good, like the way I wrote it now?